lib/travis/api/v3/services/repository/star.rb

```diff
@@ -0,0 +1,15 @@
+module Travis::API::V3
+  class Services::Repository::Star < Service
+    def run!
+      raise LoginRequired unless access_control.logged_in? or access_control.full_access?
+      raise NotFound      unless repository = find(:repository)
+      check_access(repository)
+      current_user = access_control.user
+      query.star(repository, current_user)
```

Comment:
I think the repo doesn't need to be handed in, as the query can use `find` internally.